code/modules/mob/living/carbon/xenomorph/castes/gorger/abilities_gorger.dm

```diff
@@ -241,46 +241,55 @@
 	return can_use_ability(target, TRUE)
 
 // ***************************************
-// *********** Rejuvenate
+// *********** oppose
 // ***************************************
-#define REJUVENATE_MISCLICK_CD "rejuvenate_misclick"
-/datum/action/ability/activable/xeno/rejuvenate
-	name = "Rejuvenate"
+
+/datum/action/ability/activable/xeno/oppose
+	name = "oppose"
 	action_icon_state = "rejuvenation"
-	desc = "Drains blood continuosly, slows you down and reduces damage taken, while restoring some health over time. Cancel by activating again."
-	cooldown_duration = 4 SECONDS
-	ability_cost = GORGER_REJUVENATE_COST
-	target_flags = ABILITY_MOB_TARGET
+	desc = "Violently suffuse the nearby ground with stored blood, staggering nearby marines and healing nearby xenomorphs."
+	cooldown_duration = 30 SECONDS
+	ability_cost = GORGER_OPPOSE_COST
 	keybinding_signals = list(
-		KEYBINDING_NORMAL = COMSIG_XENOABILITY_REJUVENATE,
+		KEYBINDING_NORMAL = COMSIG_XENOABILITY_OPPOSE,
 	)
 	keybind_flags = ABILITY_KEYBIND_USE_ABILITY
-	use_state_flags = ABILITY_USE_STAGGERED
 
-/datum/action/ability/activable/xeno/rejuvenate/can_use_ability(atom/A, silent, override_flags)
-	. = ..()
-	if(!.)
-		return
-	if(TIMER_COOLDOWN_CHECK(owner, REJUVENATE_MISCLICK_CD))
-		return FALSE
-
-/datum/action/ability/activable/xeno/rejuvenate/use_ability(atom/A)
+/datum/action/ability/activable/xeno/oppose/use_ability(atom/A)
 	. = ..()
 	var/mob/living/carbon/xenomorph/owner_xeno = owner
-	if(owner_xeno.has_status_effect(STATUS_EFFECT_XENO_REJUVENATE))
-		owner_xeno.remove_status_effect(STATUS_EFFECT_XENO_REJUVENATE)
-		add_cooldown()
-		return
-	owner_xeno.apply_status_effect(STATUS_EFFECT_XENO_REJUVENATE, GORGER_REJUVENATE_DURATION, owner_xeno.maxHealth * GORGER_REJUVENATE_THRESHOLD)
-	to_chat(owner_xeno, span_notice("We tap into our reserves for nourishment, our carapace thickening."))
+	add_cooldown()
 	succeed_activate()
-	TIMER_COOLDOWN_START(owner_xeno, REJUVENATE_MISCLICK_CD, 1 SECONDS)
 
-/datum/action/ability/activable/xeno/rejuvenate/ai_should_use(atom/target)
+	playsound(owner_xeno.loc, 'sound/effects/bang.ogg', 25, 0)
+	owner_xeno.visible_message(span_xenodanger("[owner_xeno] smashes into the ground!"), \
+	span_xenodanger("We smash into the ground!"))
+	owner_xeno.create_stomp() //Adds the visual effect. Wom wom wom
+	for(var/mob/living/M in range(3, get_turf(owner_xeno)))
```

Comment:
you don't need to get_turf.
Also using range instead of orange(3, owner) means the owner also gets the buff, not sure if you intended that or not.

Reply:
Listen I copied this from crusher code you gotta bear with me